src/pages/workspaces/workspace/notebooks/NotebookLauncher.js

```diff
@@ -32,37 +75,231 @@ const NotebookLauncher = _.flow(
-      }, ['edit in Jupyter'])
+    h(LabeledCheckbox, {
+      checked: hidePlaygroundMessage,
+      onChange: v => setHidePlaygroundMessage(v)
+    }, [span({ style: { marginLeft: '0.5rem' } }, ['Do not show again '])])
+  ])
+}
+
+
+const PreviewHeader = ({ queryParams, cluster, readOnlyAccess, refreshClusters, notebookName, workspace, workspace: { canShare, workspace: { namespace, name, bucketName } } }) => {
+  const signal = useCancellation()
+  const { profile: { email } } = Utils.useAtom(authStore)
+  const hidePlaygroundModal = BrowserStorage.getLocalPref('hidePlaygroundMessage')
+  const [fileInUseOpen, setFileInUseOpen] = useState(false)
+  const [playgroundModalOpen, setPlaygroundModalOpen] = useState(false)
+  const [locked, setLocked] = useState(false)
+  const [lockedBy, setLockedBy] = useState(null)
+  const [exportingNotebook, setExportingNotebook] = useState(false)
+  const [copyingNotebook, setCopyingNotebook] = useState(false)
+  const clusterStatus = cluster && cluster.status
+  const mode = queryParams['mode']
+  const notebookLink = Nav.getLink('workspace-notebook-launch', { namespace, name, notebookName })
+
+  const checkIfLocked = withErrorReporting('Error checking notebook lock status', async () => {
+    const { metadata: { lastLockedBy, lockExpiresAt } } = await Ajax(signal).Buckets.notebook(namespace, bucketName, notebookName.slice(0, -6)).getObject()
+    const hashedUser = await notebookLockHash(bucketName, email)
+    const lockExpirationDate = new Date(parseInt(lockExpiresAt))
+
+    if (lastLockedBy && (lastLockedBy !== hashedUser) && (lockExpirationDate > Date.now())) {
+      setLocked(true)
+      setLockedBy(lastLockedBy)
+    }
+  })
+
+  Utils.useOnMount(() => { checkIfLocked() })
+
+  const chooseMode = mode => {
+    Nav.history.replace({ search: '?' + qs.stringify({ mode }) })
+  }
+
+  return div({ style: { display: 'flex', alignItems: 'center', borderBottom: `2px solid ${colors.dark(0.2)}`, height: '3.5rem' } }, [
+    div({ style: { fontSize: 18, fontWeight: 'bold', backgroundColor: colors.dark(0.2), paddingRight: '4rem', paddingLeft: '4rem', height: '100%', display: 'flex', alignItems: 'center' } },
+      ['PREVIEW (READ-ONLY)']),
+    !readOnlyAccess && Utils.cond(
+      [
+        !mode, () => h(Fragment, [
+          h(ButtonSecondary, {
+            style: { paddingRight: '1rem', paddingLeft: '1rem', paddingTop: '1rem', paddingBottom: '1rem', backgroundColor: colors.dark(0.1), height: '100%', marginRight: '2px' },
+            onClick: () => locked ? setFileInUseOpen(true) : chooseMode('edit')
+          }, [icon('lock', { style: { paddingRight: '3px' } }), locked ? 'EDIT (IN USE)' : 'EDIT']),
+          h(ButtonSecondary, {
+            style: { paddingRight: '1rem', paddingLeft: '1rem', backgroundColor: colors.dark(0.1), height: '100%', marginRight: '2px' },
+            onClick: () => {
+              hidePlaygroundModal === 'true' ? chooseMode('playground') : setPlaygroundModalOpen(true)
```

Comment:
If you take the previous suggestion to just save the boolean, this should just be `!!hidePlaygroundModal`.